pennylane/labs/resource_estimation/resources_base.py

```diff
@@ -0,0 +1,384 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Base class for storing resources."""
+from __future__ import annotations
+
+import copy
+from collections import defaultdict
+from decimal import Decimal
+
+from pennylane.labs.resource_estimation.qubit_manager import QubitManager
+
+
+class Resources:
+    r"""A container to track and update the resources used throughout a quantum circuit.
+    The resources tracked include number of gates, number of wires, and gate types.
+
+    Args:
+        qubit_manager (QubitManager): A qubit tracker class which contains the number of available
+            work wires, categorized as clean, dirty or algorithmic wires.
+        gate_types (dict): A dictionary storing operations (ResourceOperator) as keys and the number
```

Comment:
```suggestion
        gate_types (dict): A dictionary storing operations (:class:`~.ResourceOperator`) as keys and the number
```

Not sure the best way to refer to the labs Classes, but the equivalent to that might be useful here.